### ISSUE TITLE
[BUG] sometimes NBT can't set to var

### ISSUE THREAD

**Issue by Radiation-pi** [SkriptHub/SkBee#19]
### What happened?

sometimes full NBT of entity can't set to var

### Steps to reproduce

log:

https://pastebin.com/p4yQquJq

My script:

```
on async load:
    create section with {_p},{_e} stored in {-horseDispearTask}:
        async run 1 ticks later repeating every seconds:
            set {_v1} to vector between {_p} and {_e}
            set {_v2} to vector from yaw ({_p}'s yaw) and pitch ({_p}'s pitch)
            set {-cache::%{_p}%::horse} to full nbt of {_e}
            set {-cache::%{_p}%::horseuuid} to uuid most from {_e}
            if any:
                {_v1} dot {_v2} < -5
                distance between {_p} and {_e} > 32
                {_p} is offline
            then:
                send "{@prefix} your horse go some where!" to {_p}
                wait 1 tick
                {_e}.remove()
                stop current task
            else if {_e} is not alive:
                stop current task

on dismount:
    set {_e} to (player's vehicle)
    "%{-cache::%player%::horseuuid}%" is "%uuid most from {_e}%"
    run section {-horseDispearTask} async with player,{_e}
```
    

### Expected behavior

This is a easy script to store player's horse to {-cache::xxx} and remove them after player not see them.

### SkBee Version

3.6.5

### Server Version

paper 1.21.1

### SkBee Info

[23:47:37 INFO]: --- [Server Info] ---
[23:47:37 INFO]: Server Version: 1.21.1-DEV-6baa110 (MC: 1.21.1)
[23:47:37 INFO]: Skript Version: 2.10.0-beta1
[23:47:37 INFO]: Skript Addons:
[23:47:37 INFO]: - skript-placeholders v1.7.0
[23:47:37 INFO]: - skript-itemsadder v1.6.1
[23:47:37 INFO]: - skript-reflect v2.5.1
[23:47:37 INFO]: - skript-db v1.4.0
[23:47:37 INFO]: - skript-worldedit v1.1.2
[23:47:37 INFO]: - RediSkript v1.3.7-SNAPSHOT
[23:47:37 INFO]: - Skript-Packet v2.2.2
[23:47:37 INFO]: - ItemsAdder v4.0.4
[23:47:37 INFO]: - MythicSkriptAddon v0.99.7
[23:47:37 INFO]: SkBee Version: 3.6.5
[23:47:37 INFO]: SkBee Website: https://github.com/SkriptHub/SkBee

**Comment by ShaneBeee** [comment]
Ok, so a few things here:
1) I spotted the issue. In my custom NBT class for entities, the getCopy method returns null if there is an error (while trying to copy the NBT from an entity), which other classes just return an empty compound, so ill fix that up to return an empty compound, and you wont get the null error.

2) I have SkBee's debug feature enabled, so I get a more verbose issue, and this is what it says:
```
[11:24:24 WARN]: com.shanebeestudios.skbee.api.nbt.NbtApiException: [?][2.14.1-SNAPSHOT]Entity NBT needs to be accessed sync!
[11:24:24 WARN]: 	at SkBee-3.6.5.jar//com.shanebeestudios.skbee.api.nbt.NBTEntity.getCompound(NBTEntity.java:80)
```
It appears NBT-API (the API SkBee uses for NBT) doesn't allow getting the nbt of an entity off the main thread (ie: async).
Theres really nothing I can do to fix that unfortunately.

So long story short, I'll fix the error, but you're going to end up getting an empty NBT compound.
You'll have to make sure to only get NBT of an entity on the main thread.

**Comment by Radiation-pi** [comment]
Thank you ShaneBeee! I'll set this in the main thread.

But I'm sometimes not sure which parts can't be run off main thread. Most of the time the server will report an error prompting me that certain operations are not allowed to be performed asynchronously, but sometimes it won't. How do I determine this?

**Comment by ShaneBeee** [comment]
> But I'm sometimes not sure which parts can't be run off main thread.

For the most part, DO NOT TOUCH THE WORLD (ie: blocks/chunks)

Touching entities off the main thread is going to be mixed. I've done lots of things with entities off the main thread (mostly getters) and they work fine. Setters on the other hand are going to probably cause issues.

That said, NBT is a whole different story due to how NBT-API handles things, and they explicitly block Entity NBT off the main thread.

**Comment by Radiation-pi** [comment]
I see..Thanks for answer my stupid question
